src/tests/cwrap/common_mock_nss_dl_load.c

```diff
@@ -0,0 +1,79 @@
+/*
+    Authors:
+        Iker Pedrosa <ipedrosa@redhat.com>
+
+    Copyright (C) 2021 Red Hat
+
+    SSSD tests: Fake nss dl load
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <dlfcn.h>
+#include <errno.h>
```

Comment:
Are those ^^ two includes still needed?

Reply:
Inclusion of `errno.h` is needed because I'm using `ENOENT`.

As a summary from `common_mock_nss_dl_load.c` I've removed the inclusion of `dlfcn.h` and `fcntl.h`. From `src/util/nss_dl_load_extra.c` I've removed `pwd.h`, `errno.h`,  `talloc.h`, `grp.h`, `db/sysdb.h`, `confdb/confdb.h`, `util/safe-format-string.h`, `util/strtonum.h`, `util/util.h` and `responder/common/responder.h`.